DependencyInjection/DoctrineExtension.php

```diff
@@ -906,13 +907,14 @@ private function createPoolCacheDefinition(ContainerBuilder $container, string $
         return $serviceId;
     }
 
-    private function getPoolNameForCacheDriver(string $driverName) : string
+    private function createArrayAdapterCachePool(ContainerBuilder $container, string $objectManagerName, string $cacheName) : string
     {
-        switch ($driverName) {
-            case 'metadata_cache':
-                return 'cache.system';
-            default:
-                return 'cache.app';
-        }
+        $id = sprintf('cache.doctrine.orm.%s.%s', $objectManagerName, str_replace('_cache', '', $cacheName));
```

Comment:
every object-manager needs separate caches, right? Otherwise there could be weird key collisions I think

Reply:
I'm not sure what the EM uses as key for the entry, but this is definitely safer than using shared caches.